tests/solvers/quadratic/gw_test.py

```diff
@@ -199,22 +199,36 @@ def reg_gw(a: jnp.ndarray, b: jnp.ndarray,
   @pytest.mark.parametrize("unbalanced", [False, True])
   def test_gw_pointcloud(self, unbalanced: bool):
     """Test basic computations pointclouds."""
-
-    def reg_gw(
-        x: jnp.ndarray, y: jnp.ndarray, a: jnp.ndarray, b: jnp.ndarray
-    ) -> float:
-      geom_x = pointcloud.PointCloud(x)
-      geom_y = pointcloud.PointCloud(y)
-      tau_a, tau_b = (self.tau_a, self.tau_b) if unbalanced else (1.0, 1.0)
-      prob = quadratic_problem.QuadraticProblem(
-          geom_x, geom_y, a=a, b=b, tau_a=tau_a, tau_b=tau_b
-      )
-      solver = gromov_wasserstein.GromovWasserstein(
-          epsilon=1.0, max_iterations=10
+    geom_x = pointcloud.PointCloud(self.x)
+    geom_y = pointcloud.PointCloud(self.y)
+    tau_a, tau_b = (self.tau_a, self.tau_b) if unbalanced else (1.0, 1.0)
+    prob = quadratic_problem.QuadraticProblem(
+        geom_x, geom_y, a=self.a, b=self.b, tau_a=tau_a, tau_b=tau_b
+    )
+    solver_ent = gromov_wasserstein.GromovWasserstein(
+        epsilon=1.0, max_iterations=10
+    )
+    solvers = [
+        solver_ent,
+    ]
+    if not unbalanced:
+      # LR Sinkhorn only implemented in balanced case.
+      solver_lr = gromov_wasserstein.GromovWasserstein(
+          rank=3, max_iterations=10
       )
-      return solver(prob).reg_gw_cost
+      solvers.append(solver_lr)
 
-    assert not jnp.isnan(reg_gw(self.x, self.y, self.a, self.b))
+    for solver in solvers:
+      out = solver(prob)
+      u = geom_x.apply_square_cost(out.matrix.sum(axis=-1)).squeeze()
+      v = geom_y.apply_square_cost(out.matrix.sum(axis=0)).squeeze()
+      c = (geom_x.cost_matrix @ out.matrix) @ geom_y.cost_matrix
+      c = (u[:, None] + v[None, :] - 2 * c)
+      if not unbalanced:
+        # Formula above is only valid for balanced case.
+        assert np.isclose(out.primal_cost, jnp.sum(c * out.matrix), rtol=1e-3)
+
+      assert not jnp.isnan(out.reg_gw_cost)
```

Comment:
Not sure if this check is necessary.

Reply:
this was here before, in this case this will be useful to check something for the unbalanced case.